lib/ds/arrayutils.flow

```diff
@@ -312,22 +312,14 @@ foldi2(a : [?], init1 : ??, init2 : ???, fn : (int, ??, ???, ?) -> Pair<??, ???>
 }
 
 replaceAtAndPad(pad : ?, xs : [?], i : int, x : ?) -> [?] {
-	l = length(xs);
-
-	if (i < l) {
-		replace(xs, i, x)
-	} else if (i == l) {
-		arrayPush(xs, x)
-	} else {
-		arrayPush(padArray(xs, pad, i - l), x)
-	}
+	replace(padArray(xs, pad, i), i, x);
 }
 
 padArray(xs : [?], pad : ?, i : int) -> [?] {
-	if (i <= 0) {
+	if (i <= length(xs)) {
 		xs
 	} else {
-		padArray(arrayPush(xs, pad), pad, i - 1);
+		padArray(arrayPush(xs, pad), pad, i);
```

Comment:
logically looks correct, but I suggest you'd better use 
`concat(xs, arrayRepeat(pad, i - length(xs))`
not to call arrayPush in recursive

Reply:
Thank you for the review!
Redid as requested.